src/java.base/share/classes/jdk/internal/foreign/Utils.java

```diff
@@ -152,68 +151,10 @@ public static MemorySegment toCString(byte[] bytes, SegmentAllocator allocator)
         return addr;
     }
 
-    /* Helper functions for offset computations. These are required so that we can avoid issuing long opcodes
-     * (e.g. LMUL, LADD) when we're operating on 'small' segments (segments whose length can be expressed with an int).
-     * C2 BCE code is very sensitive to the kind of opcode being emitted, and this workaround allows us to rescue
-     * BCE when working with small segments. This workaround should be dropped when JDK-8259609 is resolved.
-     */
-
     @ForceInline
     public static long scaleOffset(MemorySegment segment, long index, long size) {
```

Comment:
Seems like this method is pretty redundant now, and users can just do `index * size`? (at least I think the segment parameter could be dropped).